src/components/structures/UserSettings.js

```diff
@@ -357,6 +362,32 @@ module.exports = React.createClass({
             </div>);
         }
 
+        this._renderLabs = function () {
+            let features = LABS_FEATURES.map(feature => (
+                <div key={feature}>
+                    <input
+                           type="checkbox"
+                           id={feature}
+                           name={feature}
+                           defaultChecked={UserSettingsStore.isFeatureEnabled(feature)}
+                           onChange={e => UserSettingsStore.setFeatureEnabled(feature, e.target.checked)} />
+                    <label htmlFor={feature}>{feature}</label>
+                </div>
+            ));
+            return (
+                <div>
+                    <h3>Labs</h3>
+
+                    <div className="mx_UserSettings_section">
+                        <p>These experimental features may change, break or disappear at any time. We make absolutely no guarantees about what may happen if you turn one of these experiments on, and your client may even spontaneously combust. Jokes aside, your client may delete all your data or your security and privacy could be compromised in unexpected ways. Please proceed with caution.</p>
+                        {features}
+                        <br />
+                        <div className="mx_UserSettings_button" onClick={() => window.location.reload()}>Restart Vector</div>
```

Comment:
Is this necessary? I don't think we need it for the current two, and if future features need to know to update, we can dispatch through the dispatcher.